src/hooks/useFetch.js

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from 'react';
+
+const useFetch = (url) => {
+  const [data, setData] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    fetch(url)
+      .then(response => {
+        if (!response.ok) {
+          throw new Error('Squizard is sorry!  Something has gone wrong, please try again.');
+        }
+        return response.json();
+      })
+      .then(data => {
+        setData(data);
```

Comment:
It almost got us.

Reply:
Seriously, good catch! I would not have thought of that.